src/cls/queue/cls_queue_src.cc

```diff
@@ -332,7 +332,8 @@ int queue_list_entries(cls_method_context_t hctx, const cls_queue_list_op& op, c
     uint64_t entry_start_offset = start_offset - bl.length();
     CLS_LOG(20, "INFO: queue_list_entries(): Entry start offset accounting for leftover data is %lu", entry_start_offset);
     bl.claim_append(bl_chunk);
-    bl_chunk = std::move(bl);
+    bl_chunk = bl;
+    bl.clear();
```

Comment:
are we actually seeing bugs related to this?

`bufferlist`'s move operations do call `clear()` on their operand, but it's fine to add another `bl.clear()` just to be safe. we shouldn't need to change this move into a copy though, since that adds unnecessary atomics to inc/dec the reference counts

Reply:
the question is whether one can use bl after it is moved from

Reply:
you can, yes. accessing a moved-from object is not undefined behavior

from https://en.cppreference.com/w/cpp/utility/move:
> Unless otherwise specified, all standard library objects that have been moved from are placed in a "valid but unspecified state", meaning the object's class invariants hold

this 'valid but unspecified' guarantee should be the bare minimum for our own types, though we're free to offer additional guarantees

for a generic type `T`, we don't know whether or not `T` supports move construction/assignment so `T a = std::move(b);` could either result in a copy or a move. since we don't know `b`'s state, we'd want to do something like `b = T{}` or `b.clear()` before reusing it

for `bufferlist`, we can see from its implementation what the moved-from state is, but it's worth adding the `bl.clear()` just to be safe

Reply:
ok, thanks for clarifying, I believed that it was not permitted.  The problem here is Pritha's change was hoping to address a behavior that's been reported, where the OSD suicides, due to this CLS op running too long.  I haven't spotted a clear reason that has to happen.  Pritha believed the time was being spent in the copy operation.

Reply:
I'm with Casey here. moving from should work fine. In this context it probably doesn't save us much from the atomics, but it'd be very surprising to me if it were causing the bug.